rundeckapp/grails-app/services/rundeck/services/NodeStatusService.groovy

```diff
@@ -151,78 +207,98 @@ class NodeStatusService implements RundeckProjectConfigurable, InitializingBean
             frameworkProject(project)
             storageTree(storageService.storageTreeWithContext(authContext))
             nodeService(nodeService)
+            executionListener(executionListenerWf)
         }
         builder.framework(framework)
         builder.authContext(authContext)
         builder.threadCount(1)
-        builder.executionListener(executionListener)
 
         def checkCommand
 
         //TODO: we can use a node attribute to get the healtcheck command
         if ("windows".equalsIgnoreCase(node.osFamily)) {
-            checkCommand="dir"
-        }else{
-            checkCommand="uname"
+            checkCommand = "dir"
+        } else {
+            checkCommand = "uname"
         }
 
+        sysThreadBoundOut.installThreadStream(loggingService.createLogOutputStream(
+                new NoLogWriter(),
+                LogLevel.DEBUG,
+                contextmanager,
+                logOutFlusher,
+                null
+        ))
+        sysThreadBoundErr.installThreadStream(loggingService.createLogOutputStream(
+                new NoLogWriter(),
+                LogLevel.ERROR,
+                contextmanager,
+                logErrFlusher,
+                null
+        ))
+
         def executorReachable
         def executorTimeout
         def statusDescription
 
         try {
-            CommandExec step = new CommandExec(adhocRemoteString: checkCommand,
-                                               adhocExecution: true)
+            CommandExec step = new CommandExec(
+                    adhocRemoteString: checkCommand,
+                    adhocExecution: true
+            )
             StepExecutionItem item = executionUtilService.itemForWFCmdItem(step, null, null)
 
             def context = builder.build()
 
             NodeStepExecutor interpreter = framework.getNodeStepExecutorForItem(item);
             def result = interpreter.executeNodeStep(context, item, node);
 
-            if(result.isSuccess()){
+            if (result.isSuccess()) {
                 executorReachable = "successful"
-                executorTimeout="false"
+                executorTimeout = "false"
                 statusDescription = "Health Check successful"
-            }else{
-                executorReachable="failure"
-                executorTimeout="true"
+            } else {
+                executorReachable = "failure"
+                executorTimeout = "true"
                 statusDescription = result.failureReason
             }
 
-        }catch(Exception e){
-            executorReachable="fail"
-            executorTimeout="false"
+        } catch (Exception e) {
+            executorReachable = "fail"
+            executorTimeout = "false"
             statusDescription = e.message
         }
```

Comment:
since you are binding the thread outputstream earlier, i think you should add `finally{` section to reset the threadbound outputstream. (see how it is done here https://github.com/rundeck/rundeck/blob/master/rundeckapp/grails-app/services/rundeck/services/ExecutionService.groovy#L1153-L1156 ) in that case it only happens on exception because the thread is handled later when it completes, but in your case i think you want to do that reset inside `finally`